### PR TITLE
Refactor accordion so it uses details/summary API

### PR DESCRIPTION
**Pull request by gsi-kevincarrera**

### CODE REVIEW

**Comment by markteekman** [comment]
Thanks for your contribution again @gsi-kevincarrera! I'll take a look at it soon 😄

**Comment by gsi-kevincarrera** [comment]
Willing to get your feedback here :)

**Comment by markteekman** [comment]
@gsi-kevincarrera thanks again for taking the time to look at this! I've got a couple of concerns before we can merge this to main, especially since it's a breaking change with AccordionItem.astro being deleted and refactoring of existing themes is needed if anyone updates to the latest version of the package. Currently there's a big difference in the look and feel of the 'old' vs the new accordions:

**Old** (general design)
Scherm­afbeelding 2024-02-03 om 09 52 02

**New** (general design)
Scherm­afbeelding 2024-02-03 om 09 52 18

**Old** (focus outline)
Scherm­afbeelding 2024-02-03 om 09 52 40

**New** (focus outline)
Scherm­afbeelding 2024-02-03 om 09 52 29

Some other concerns I had:
- In your style tag you use the `is:global` directive, this causes the styles to be applied multiple times when an accordion item is on the page. I think we can leave it out. 
- For the classes, all the components use [BEM](https://getbem.com/introduction/) so these should be renamed to `accordion-container` and `plus-icon`. 
- The `name` attirbute on `details` is currently not supported by Firefox, but can we a progressive enhancement. I would just name the prop `name` though. 
- All components need to support TypeScript, so you need to have an interface that defines the prop types
- All components need to use `rem` instead of `px` units
- It would be preferable to keep Accordion.astro as well as AccordionItem.astro so we can prevent breaking changes to themes updating to the latest version.

I think if we tackle these issues and make sure the component looks the same then we are there 🙂

File: Accordion.astro [hunk 1]
```diff
@@ -1,126 +1,71 @@
...
+  }
 
-<style is:global>
-  .accordion__wrapper {
-    list-style: none;
-    padding: 0;
+  details[open] > *:not(summary) {
+    animation: details-show .15s ease-out;
   }
+
```
Comment (markteekman) [inline Accordion.astro:69]: Last line can be removed

File: Accordion.astro [hunk 2]
```diff
@@ -1,126 +1,71 @@
...
-
-  document.addEventListener('keydown', (event) => {
-    const accordionItem = event.target.closest('.accordion__item')
-
-    if (event.key !== 'Escape') return
-    if (!accordionItem) return
-
-    if (isAccordionOpen(accordionItem)) {
-      closeAccordionItem(accordionItem)
+  @keyframes details-show {
```
Comment (markteekman) [inline Accordion.astro:60]: Keyframes may be placed at the bottom

File: Accordion.astro [hunk 3]
```diff
@@ -1,126 +1,71 @@
...
-
-    accordionItem.classList.remove('is-active')
-    accordionPanel.style.height = 0
-    accordionItemHeader.focus()
-    accordionItemHeader.setAttribute('aria-expanded', false)
-    accordionToggleIndicator.innerHTML = `<svg class="header__toggle-indicator" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"/></svg>`
+  summary {
+    cursor: pointer;
+    font-weight: bold;
+    list-style-type: none;
```
Comment (markteekman) [inline Accordion.astro:45]: It's not a list item so you can remove this

**Review by markteekman — CHANGES_REQUESTED** [review]

**Comment by gsi-kevincarrera** [comment]
Thanks for your time, I will be working in all of them, i ll come back soon

**Comment by markteekman** [comment]
@gsi-kevincarrera how's it going, need any help with this? 🙂

**Comment by gsi-kevincarrera** [comment]
Sorry,  I've been really busy, but thanks for the reminder 😅 I'm going to make time to finish the task

**Comment by markteekman** [comment]
No worries Kevin and no rush! Was just checking whether you still wanted to do this, take your time 😊

**Comment by markteekman** [comment]
Hey @gsi-kevincarrera, this PR has gone a little stale, which is okay, but since I've got a couple of changes I'm going to make to this project I'll have to close this PR down. I do want to thank you for the time you've invested in this 😃

**Comment by gsi-kevincarrera** [comment]
> Hey @gsi-kevincarrera, this PR has gone a little stale, which is okay, but since I've got a couple of changes I'm going to make to this project I'll have to close this PR down. I do want to thank you for the time you've invested in this 😃 

I do really want to apologize for not being able to complete it. It's been a tough year, with a lot on my plate, thank you

**Comment by markteekman** [comment]
gsi-kevincarrera there's truly no need to apologize, no worries 🙂

**Pull request closed without merging**